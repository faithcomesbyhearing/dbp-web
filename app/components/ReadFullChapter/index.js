/**
 *
 * ReadFullChapter
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Link from 'next/link';
import messages from './messages';

function ReadFullChapter({ activeTextId, activeBookId, activeChapter }) {
	const textId = String(activeTextId || '').toLowerCase();
	const bookId = String(activeBookId || '').toLowerCase();
	const chapter = activeChapter;

	const href = `/app?bibleId=${textId}&bookId=${bookId}&chapter=${chapter}`;
	const as = `/bible/${textId}/${bookId}/${chapter}`;
	return (
		<div className={'read-chapter-container'}>
			<Link legacyBehavior href={href} as={as}>
				<button type={'button'} className={'read-chapter'}>
					<FormattedMessage {...messages.readFullChapter} />
				</button>
			</Link>
		</div>
	);
}

ReadFullChapter.propTypes = {
	activeTextId: PropTypes.string,
	activeBookId: PropTypes.string,
	activeChapter: PropTypes.number,
};

ReadFullChapter.defaultProps = {
	activeTextId: '',
	activeBookId: '',
	activeChapter: 1,
};

export default ReadFullChapter;
