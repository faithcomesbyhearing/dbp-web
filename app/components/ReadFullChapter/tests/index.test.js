import React from 'react';
import { render } from '@testing-library/react';
import ReadFullChapter from '..';

jest.mock('react-intl', () => ({
	FormattedMessage: ({ defaultMessage }) => <span>{defaultMessage}</span>,
	defineMessages: (messages) => messages,
}));

const props = {
	activeTextId: 'ENGESV',
	activeBookId: 'GEN',
	activeChapter: 1,
};

describe('ReadFullChapter component', () => {
	it('should match previous snapshot', () => {
		const { asFragment } = render(<ReadFullChapter {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
