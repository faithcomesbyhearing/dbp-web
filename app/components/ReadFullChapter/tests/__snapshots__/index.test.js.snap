// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadFullChapter component should match previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="read-chapter-container"
  >
    <button
      class="read-chapter"
      type="button"
    >
      <span>
        Read Full Chapter
      </span>
    </button>
  </div>
</DocumentFragment>
`;
